### PR TITLE
Update plt.figure() docstring.

### PR DESCRIPTION
**Pull request by anntzer**
- Don't suggest using backends for customization (it's quite heavy
  handed anyways, and we may soon have figure hooks to simplify that).
- Replace tight_layout/constrained_layout by layout engines (none of the
  kwargs appear explicitly in the signature, but because figure() is
  such a central entry point, it seems reasonable to be a bit more wordy
  in the descriptions).

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/pyplot.py [hunk 1]
```diff
@@ -712,40 +712,32 @@ def figure(num=None,  # autoincrement if None, else integer from 1-N
...
-        elements.  Like ``tight_layout``, but designed to be more
-        flexible.  See
-        :doc:`/tutorials/intermediate/constrainedlayout_guide`
-        for examples.  (Note: does not work with `add_subplot` or
-        `~.pyplot.subplot2grid`.)
-
+    layout : {'constrained', 'tight', `.LayoutEngine`, None}, default: None
+        The layout mechanism for positioning of plot elements to avoid
+        overlapping Axes decorations (labels, ticks, etc). Note that layout
+        managers can have significant performance penalties. Defaults to
```
Comment (timhoffm) [inline lib/matplotlib/pyplot.py:724]: Sounds quite technical. How about

> can make the showing of a figure measurably slower.

Comment (timhoffm) [inline lib/matplotlib/pyplot.py:724]: Sounds quite technical. How about

> can make the showing of a figure measurably slower.

Comment (timhoffm) [inline lib/matplotlib/pyplot.py:724]: Sounds quite technical. How about

> can make the showing of a figure measurably slower.

Comment (timhoffm) [inline lib/matplotlib/pyplot.py:724]: Sounds quite technical. How about

> can make the showing of a figure measurably slower.

Comment (timhoffm) [inline lib/matplotlib/pyplot.py:724]: Sounds quite technical. How about

> can make the showing of a figure measurably slower.

Comment (timhoffm) [inline lib/matplotlib/pyplot.py:724]: Sounds quite technical. How about

> can make the showing of a figure measurably slower.

Comment (anntzer) [reply lib/matplotlib/pyplot.py]: Went for a variation on that.

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/pyplot.py [hunk 2]
```diff
@@ -712,40 +712,32 @@ def figure(num=None,  # autoincrement if None, else integer from 1-N
...
+    layout : {'constrained', 'tight', `.LayoutEngine`, None}, default: None
+        The layout mechanism for positioning of plot elements to avoid
+        overlapping Axes decorations (labels, ticks, etc). Note that layout
+        managers can measurably slow down figure display. Defaults to *None*
+        (but see the documentation of the `.Figure` constructor regarding the
+        interaction with rcParams).
 
     **kwargs : optional
-        See `~.matplotlib.figure.Figure` for other possible arguments.
+        Additional kwargs will be passed to the `.Figure` constructor.
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py:729]:
```suggestion
        Additional keyword arguments will be passed to the `.Figure` constructor.
```

Comment (anntzer) [reply lib/matplotlib/pyplot.py]: yes

File: lib/matplotlib/pyplot.py [hunk 3]
```diff
@@ -712,40 +712,32 @@ def figure(num=None,  # autoincrement if None, else integer from 1-N
...
     `~matplotlib.figure.Figure`
-        The `.Figure` instance returned will also be passed to
-        new_figure_manager in the backends, which allows to hook custom
-        `.Figure` classes into the pyplot interface. Additional kwargs will be
-        passed to the `.Figure` init function.
 
     Notes
     -----
+    Newly created figures will be passed to the ``new_figure_manager`` function
+    provided by the current backend, which will install a canvas and a manager
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/pyplot.py]: Sure, linked https://matplotlib.org/stable/api/backend_template_api.html#matplotlib.backends.backend_template.new_figure_manager which I think is the best for now.

**Pull request merged**